packages/vite/src/node/server/transformRequest.ts

```diff
@@ -168,7 +170,50 @@ async function loadAndTransform(
 
   // load
   const loadStart = isDebug ? performance.now() : 0
-  const loadResult = await pluginContainer.load(id, { ssr })
+  let loadResult: LoadResult
+
+  const idWithoutHmrFlag = id.replace(DEP_VERSION_RE, '')
+  const loadCacheKey = ssr ? `ssr:${idWithoutHmrFlag}` : idWithoutHmrFlag
+
+  loadResult = await pluginContainer.load(id, { ssr })
+
+  const includedInPersistentCache =
+    _persistentCache &&
+    (!server.config.resolvedServerPersistentCacheOptions?.exclude ||
+      !server.config.resolvedServerPersistentCacheOptions.exclude(url))
+
+  // Persist load result just in case (for example: svelte component CSS subrequest)
+
+  if (
+    _persistentCache &&
+    includedInPersistentCache &&
+    idWithoutHmrFlag !== file
+  ) {
+    const fileCacheInfo = (_persistentCache.manifest.files[file] =
+      _persistentCache.manifest.files[file] ?? {
+        relatedModules: {}
+      })
+    if (loadResult != null) {
+      const saveKey = _persistentCache.getKey(loadCacheKey)
+      let code: string
+      let map: any | null
+      if (typeof loadResult === 'string') {
+        code = loadResult
+      } else {
+        code = loadResult.code
+        map = loadResult.map
+      }
+      await _persistentCache.write(saveKey, id, null, ssr, file, code, map)
+      fileCacheInfo.relatedModules[loadCacheKey] = saveKey
+      _persistentCache.queueManifestWrite()
+    } else {
+      const saveKey = fileCacheInfo.relatedModules[loadCacheKey]
+      if (saveKey) {
+        loadResult = await _persistentCache.read(saveKey)
```

Comment:
Can you explain a bit more the goal of this whole block? (maybe with an example of two ids for svelte)

I don't understand why the override of `await pluginContainer.load(id, { ssr })` is only done when `idWithoutHmrFlag !== file`

Reply:
The goal is to restore the result of `plugin.load` in case it needed side effects from `plugin.transform`. For example (not real requests):
- `transform`: `Foo.svelte` => put `<style>` part into an internal cache as side-effect
- `load`: `Foo.svelte?type=css` => retrieve the content of `<style>` from the internal cache.

Reply:
I will bootstrap the a simple svelte template with the inspect plugin to better understand how it works and comeback here after.

Reply:
As I understand it, the `load` hook will return nothing for `Foo.svelte?type=css` when `Foo.svelte` was already cached (and thus, never processed by `transform` pipeline). By caching the `load` result, Vite can still access the `Foo.svelte?type=css` contents for the cached `Foo.svelte` module.

Since the `load` hook is always called (even for cached modules), this is a necessary workaround.

Reply:
Ideally, there would be a way for a `load` result to declare a "dependency" on another file's contents. This way, Vite wouldn't have to cache the `load` result of *every* file.